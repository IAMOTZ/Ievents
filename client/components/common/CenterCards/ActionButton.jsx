import React from 'react';
import { Link } from 'react-router-dom';

const ActionButton = (props) => {
  let component;
  if (props.isAdmin) {
    component = (
      <Link
        to="/editCenter"
        class="btn mt-3 ie-dark-button"
        id={props.id}
        onClick={props.startEditingCenter}
      >Edit
      </Link>
    );
  } else {
    component = (
      <button
        className="btn mt-3 ie-dark-button"
        data-toggle="modal"
        data-target="#center-details-modal"
        id={props.id}
        onClick={props.showCenterDetails}
      >Details
      </button>
    );
  }
  return component;
};

export default ActionButton;
