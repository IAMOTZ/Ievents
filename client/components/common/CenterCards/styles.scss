@import "../../../styles/variables";

.centerCard {
  margin: 5px;
  .card-img-top {
    height: 250px;
  }
  .badge-container {
    display: flex;
    width: 100%;
    margin-top: -50px;
    justify-content: flex-end;
    padding-right: 20px;
    .seat-badge {
      font-size: 20px;
      background-color: rgba(0,0,0,0.4);
    }
  }
  .icons {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
  }
  .admin-only-icons {
    display: flex;
    justify-content: center;
    margin-top: 15px;
    span {
      cursor: pointer;
    }
    i {
      margin-top: 3px;
    }
  }
  .icons, .admin-only-icons {
    span, a {
      color: grey;
      font-size: 15px;
    }
  }
  .center-details {
    font-size: 15px;
  }
}
