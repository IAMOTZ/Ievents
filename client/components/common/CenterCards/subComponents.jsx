import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export const RenderCenterBody = (props) => {
  const { center } = props;
  return (
    <div className="card-body d-flex flex-column">
      <div>
        <h4 className="card-title text-capitalize">{center.name}</h4>
        <div className="icons">
          <div>
            <i className="fa fa-map-marker fa-fw" aria-hidden="true" />&nbsp;
            <span className="text-capitalize">{center.location}</span>
          </div>
          <div>
            <i className="fa fa-info-circle fa-fw" aria-hidden="true" />&nbsp;
            <a
              className="no-outline-btn"
              onClick={() => props.createModalContent(center)}
              data-toggle="modal"
              href="#center-details-modal"
              role="button"
            >About center
            </a>
          </div>
        </div>
      </div>
      {
        props.isAdmin ?
          <div className="admin-only-icons">
            <i className="fa fa-pencil fa-fw" aria-hidden="true" />&nbsp;
            <Link
              to={`/centers/${center.id}/edit`}
              id={center.id}
              onClick={props.onEdit}
            >Edit Center
            </Link>
          </div> : null
      }
      <Link
        to="/addEvent"
        onClick={props.onBook}
        id={center.id}
        className="btn ie-blue-button mt-2 btn-block"
      > Book now for â‚¦{center.price}/Day
      </Link>
    </div>
  );
};
/* eslint-disable react/forbid-prop-types */
RenderCenterBody.propTypes = {
  center: PropTypes.object.isRequired,
  createModalContent: PropTypes.func.isRequired,
  isAdmin: PropTypes.bool,
  onBook: PropTypes.func.isRequired,
  onEdit: PropTypes.func,
};
RenderCenterBody.defaultProps = {
  onEdit: () => { },
  isAdmin: false,
};

export const RenderCenterBodyInTransactions = props => (
  <div className="card-body d-flex flex-column">
    <div>
      <h4 className="card-title text-capitalize">{props.center.name}</h4>
      <div>
        <i className="fa fa-map-marker fa-fw" aria-hidden="true" />&nbsp;
        <span className="text-capitalize">{props.center.location}</span>
      </div>
    </div>
    <Link
      onClick={props.onViewTransactions}
      to={`/transaction/${props.center.id}/events`}
      id={props.center.id}
      className="btn ie-blue-button mt-2 btn-block"
    >View Transactions
    </Link>
  </div>
);
RenderCenterBodyInTransactions.propTypes = {
  center: PropTypes.object.isRequired,
  onViewTransactions: PropTypes.func.isRequired,
};

