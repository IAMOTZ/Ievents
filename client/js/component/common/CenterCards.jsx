import React from 'react';
import { Link } from 'react-router-dom';

const defaultImage = '../../../images/defaultImgx4.jpeg';

const CenterCards = (props) => {
  return props.centers.map((center) => (
    <div className="col-lg-4 col-md-6 col-sm-12" key={center.id}>
      <div className="card">
        <div>
          <img className="card-img-top" src={center.images ? center.images[0] : defaultImage} alt="" />
          <span className="badge text-white p-2 seat-badge ">{center.capacity} seats</span>
        </div>
        <div className="card-body d-flex flex-column">
          <div>
            <h4 className="card-title text-capitalize">{center.name}</h4>
            <i className="fa fa-map-marker fa-fw" aria-hidden="true"></i>&nbsp; <span className="text-capitalize">{center.location}</span>
          </div>
          <ActionBtn isAdmin={props.isAdmin || false}
            id={center.id}
            editAction={props.editAction}
            btnAction={props.btnAction} />
        </div>
      </div>
    </div>
  ));
}

const ActionBtn = (props) => {
  if (props.isAdmin) {
    return (
      <Link to="/editCenter"
        class="btn text-white mt-3"
        id={props.id}
        onClick={props.editAction}>Edit</Link>
    );
  } else {
    return (
      <a role="button"
        class="btn text-white mt-3"
        data-toggle="modal"
        data-target="#center-details-modal"
        id={props.id}
        onClick={props.btnAction}>Details</a>
    );
  }
}

export default CenterCards;
