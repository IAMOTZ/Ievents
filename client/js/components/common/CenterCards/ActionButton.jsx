import React from 'react';
import { Link } from 'react-router-dom';

const ActionButton = (props) => {
  let component;
  if (props.isAdmin) {
    component = (
      <Link
        to="/editCenter"
        class="btn text-white mt-3"
        id={props.id}
        onClick={props.startEditingCenter}
      >Edit
      </Link>
    );
  } else {
    component = (
      <button
        className="btn text-white mt-3"
        data-toggle="modal"
        data-target="#center-details-modal"
        id={props.id}
        onClick={props.showCenterDetails}
      >Details
      </button>
    );
  }
  return component;
};

export default ActionButton;
