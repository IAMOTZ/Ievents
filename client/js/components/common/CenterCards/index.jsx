import React from 'react';
import ActionButton from './ActionButton';

const defaultImage = '/images/defaultImgx4.jpeg';

const CenterCards = props => props.centers.map(center => (
  <div className="col-lg-4 col-md-6 col-sm-12" key={center.id}>
    <div className="card">
      <div>
        <img className="card-img-top" src={center.images ? center.images[0] : defaultImage} alt="" />
        <span className="badge text-white p-2 seat-badge ">{center.capacity} seats</span>
      </div>
      <div className="card-body d-flex flex-column">
        <div>
          <h4 className="card-title text-capitalize">{center.name}</h4>
          <i className="fa fa-map-marker fa-fw" aria-hidden="true" />&nbsp;
          <span className="text-capitalize">{center.location}</span>
        </div>
        <ActionButton
          isAdmin={props.isAdmin || false}
          id={center.id}
          startEditingCenter={props.editAction}
          showCenterDetails={props.btnAction}
        />
      </div>
    </div>
  </div>
));

export default CenterCards;
