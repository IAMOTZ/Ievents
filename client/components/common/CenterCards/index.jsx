import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './styles.scss';

const defaultImage = '/images/defaultImgx4.jpeg';

const CenterCards = props => props.centers.map(center => (
  <div className="col-lg-4 col-md-6 col-sm-1" key={center.id}>
    <div className="card centerCard">
      <div>
        <img className="card-img-top" src={center.images ? center.images[0] : defaultImage} alt="The Center" />
        <div className="badge-container">
          <span className="badge text-white p-2 seat-badge ">{center.capacity} seats</span>
        </div>
      </div>
      <div className="card-body d-flex flex-column">
        <div>
          <h4 className="card-title text-capitalize">{center.name}</h4>
          <div className="icons">
            <div>
              <i className="fa fa-map-marker fa-fw" aria-hidden="true" />&nbsp;
              <span className="text-capitalize">{center.location}</span>
            </div>
            <div>
              <i className="fa fa-info-circle fa-fw" aria-hidden="true" />&nbsp;
              <a data-toggle="collapse" href={`#center-details-collapse-${center.id}`} role="button">About center</a>
            </div>
          </div>
          <div className="collapse" id={`center-details-collapse-${center.id}`}>
            <br />
            <span className="center-details">{center.details}</span>
          </div>
        </div>
        {
          props.isAdmin ?
            <div className="admin-only-icons">
              <i className="fa fa-pencil fa-fw" aria-hidden="true" />&nbsp;
              <Link
                to="/editCenter"
                id={center.id}
                onClick={props.onEdit}
              >Edit Center
              </Link>
            </div> : null
        }
        <Link
          to="/addEvent"
          onClick={props.onBook}
          id={center.id}
          className="btn ie-blue-button mt-2 btn-block"
        > Book now for â‚¦{center.price}/Day
        </Link>
      </div>
    </div>
  </div>
));

/* eslint-disable react/forbid-prop-types */
CenterCards.propTypes = {
  centers: PropTypes.array.isRequired,
  onBook: PropTypes.func.isRequired,
  onEdit: PropTypes.func,
};

export default CenterCards;
