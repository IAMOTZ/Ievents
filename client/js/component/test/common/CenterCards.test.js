import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import CenterCards from '../../common/CenterCards.jsx';

const props = {
  centers: [
    {
      id: 1,
      name: 'test-center1',
      location: 'test-center-location',
      details: 'test-center-details',
      capacity: '200',
      price: '500',
      images: ['http://testImage.jpg'],
      bookedOn: ['02/12/2016', '02/12/2017', '02/12/2018'],
    },
  ],
  isAdmin: false,
  editAction: () => { },
  btnAction: () => { },
};

const alterProps = newProps => ({
  centers: [...newProps.centers],
  isAdmin: Boolean(newProps.isAdmin),
  editAction: () => { },
  btnAction: () => { },
});

describe('<CenterCards />', () => {
  it('should render correctly if it is a normal user', () => {
    const tree = renderer.create(
      <Router>
        <CenterCards {...props} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly if it is an admin user', () => {
    const newProps = alterProps({
      centers: props.centers,
      isAdmin: true,
    });
    const tree = renderer.create(
      <Router>
        <CenterCards {...newProps} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render with the default image if no image was passed in', () => {
    const newProps = alterProps({
      centers: [{ ...props.centers, images: null }],
    });
    const tree = renderer.create(
      <Router>
        <CenterCards {...newProps} />
      </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
