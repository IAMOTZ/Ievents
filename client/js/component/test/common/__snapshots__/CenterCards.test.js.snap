// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterCards /> should render correctly if it is a normal user 1`] = `
<div
  className="col-lg-4 col-md-6 col-sm-12"
>
  <div
    className="card"
  >
    <div>
      <img
        alt=""
        className="card-img-top"
        src="http://testImage.jpg"
      />
      <span
        className="badge text-white p-2 seat-badge "
      >
        200
         seats
      </span>
    </div>
    <div
      className="card-body d-flex flex-column"
    >
      <div>
        <h4
          className="card-title text-capitalize"
        >
          test-center1
        </h4>
        <i
          aria-hidden="true"
          className="fa fa-map-marker fa-fw"
        />
         
        <span
          className="text-capitalize"
        >
          test-center-location
        </span>
      </div>
      <button
        className="btn text-white mt-3"
        data-target="#center-details-modal"
        data-toggle="modal"
        id={1}
        onClick={[Function]}
      >
        Details
      </button>
    </div>
  </div>
</div>
`;

exports[`<CenterCards /> should render correctly if it is an admin user 1`] = `
<div
  className="col-lg-4 col-md-6 col-sm-12"
>
  <div
    className="card"
  >
    <div>
      <img
        alt=""
        className="card-img-top"
        src="http://testImage.jpg"
      />
      <span
        className="badge text-white p-2 seat-badge "
      >
        200
         seats
      </span>
    </div>
    <div
      className="card-body d-flex flex-column"
    >
      <div>
        <h4
          className="card-title text-capitalize"
        >
          test-center1
        </h4>
        <i
          aria-hidden="true"
          className="fa fa-map-marker fa-fw"
        />
         
        <span
          className="text-capitalize"
        >
          test-center-location
        </span>
      </div>
      <a
        class="btn text-white mt-3"
        href="/editCenter"
        id={1}
        onClick={[Function]}
      >
        Edit
      </a>
    </div>
  </div>
</div>
`;

exports[`<CenterCards /> should render with the default image if no image was passed in 1`] = `
<div
  className="col-lg-4 col-md-6 col-sm-12"
>
  <div
    className="card"
  >
    <div>
      <img
        alt=""
        className="card-img-top"
        src="/images/defaultImgx4.jpeg"
      />
      <span
        className="badge text-white p-2 seat-badge "
      >
        200
         seats
      </span>
    </div>
    <div
      className="card-body d-flex flex-column"
    >
      <div>
        <h4
          className="card-title text-capitalize"
        >
          test-center1
        </h4>
        <i
          aria-hidden="true"
          className="fa fa-map-marker fa-fw"
        />
         
        <span
          className="text-capitalize"
        >
          test-center-location
        </span>
      </div>
      <button
        className="btn text-white mt-3"
        data-target="#center-details-modal"
        data-toggle="modal"
        id={1}
        onClick={[Function]}
      >
        Details
      </button>
    </div>
  </div>
</div>
`;
